code/game/objects/items/weaponry.dm

```diff
@@ -485,6 +489,8 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 /obj/item/cane/white/proc/on_transform(obj/item/source, mob/user, active)
 	SIGNAL_HANDLER
 
+	src.active = active
+	inhand_icon_state = active ? on_inhand_icon_state : inhand_icon_state
```

Comment:
Also for future reference, this should always be done in update_icon_state(). 

The proper way to do this;
https://github.com/tgstation/tgstation/blob/5247649e64dbf8424ae9d9962a0a7f2acc6de657/code/game/objects/items/toys.dm#L449-L453